File: app/src/test/java/com/solt9029/editmasterandroid/util/CalcUtilTest.java
package com.solt9029.editmasterandroid.util;

import com.solt9029.editmasterandroid.constants.SecondConstants;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.Matchers.samePropertyValuesAs;
import static org.junit.Assert.assertThat;


public class CalcUtilTest {
    @Test
    public void calcSecondsPerNote() {
        float actual = CalcUtil.calcSecondsPerNote(240);
        float expected = 1f / 96f;
        assertThat(actual, is(expected));
    }

    @Test
    public void calcNoteIndexRangeInSecondRange() {
        IndexRange actual = CalcUtil.calcNoteIndexRangeInSecondRange(SecondConstants.Range.AUTO, 10, 150, 1);
        IndexRange expected = new IndexRange(539, 541);
        assertThat(actual, samePropertyValuesAs(expected));
    }
}

File: app/src/test/java/com/solt9029/editmasterandroid/util/CalcUtilTest.kt
package com.solt9029.editmasterandroid.util

import com.solt9029.editmasterandroid.constants.SecondConstants
import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.Matchers.samePropertyValuesAs
import org.junit.Assert.assertThat
import org.junit.Test


class CalcUtilTest {
    @Test
    fun calcSecondsPerNote() {
        val actual = CalcUtil.calcSecondsPerNote(240f)
        val expected = 1f / 96f
        assertThat(actual, `is`(expected))
    }

    @Test
    fun calcNoteIndexRangeInSecondRange() {
        val actual = CalcUtil.calcNoteIndexRangeInSecondRange(SecondConstants.RANGE_AUTO, 10f, 150f, 1f)
        val expected = IndexRange(539, 541)
        assertThat(actual, samePropertyValuesAs(expected))
    }
}
